packages/config-plugins/src/utils/Updates.ts

```diff
@@ -19,3 +25,82 @@ export function getUpdateUrl(
   }
   return `https://exp.host/@${user}/${config.slug}`;
 }
+
+export function getNativeVersion(
+  config: Pick<ExpoConfig, 'version'> & {
+    android?: Pick<Android, 'versionCode'> & TempRuntimeVersion;
+    ios?: Pick<IOS, 'buildNumber'> & TempRuntimeVersion;
+  },
+  platform: 'android' | 'ios'
+): string {
+  const version = IOSConfig.Version.getVersion(config);
+  switch (platform) {
+    case 'ios': {
+      const buildNumber = IOSConfig.Version.getBuildNumber(config);
+      return `${version}(${buildNumber})`;
+    }
+    case 'android': {
+      const versionCode = AndroidConfig.Version.getVersionCode(config);
+      return `${version}(${versionCode})`;
+    }
+    default: {
+      throw new Error(
+        `"${platform}" is not a supported platform. Choose either "ios" or "android".`
+      );
+    }
+  }
+}
+
+/**
+ * Compute runtime version policies.
+ * @return an expoConfig with only string valued platform specific runtime versions.
+ */
+export const withRuntimeVersion: ConfigPlugin = config => {
+  if (config.ios || config.runtimeVersion) {
+    config.ios = {
+      ...config.ios,
+      runtimeVersion: getRuntimeVersion(config, 'ios'),
+    } as any; //TODO(JJ) remove this cast in SDK 43 https://linear.app/expo/issue/ENG-1869/remove-tempruntimeversion-in-expoconfig
+  }
+  if (config.android || config.runtimeVersion) {
+    config.android = {
+      ...config.android,
+      runtimeVersion: getRuntimeVersion(config, 'android'),
+    } as any; //TODO(JJ) remove this cast in SDK 43 https://linear.app/expo/issue/ENG-1869/remove-tempruntimeversion-in-expoconfig
+  }
+  delete config.runtimeVersion;
+  return config;
+};
+
+export function getRuntimeVersion(
+  config: Pick<ExpoConfig, 'version' | 'sdkVersion'> &
+    TempRuntimeVersion & {
+      android?: Pick<Android, 'versionCode'> & TempRuntimeVersion;
+      ios?: Pick<IOS, 'buildNumber'> & TempRuntimeVersion;
+    },
+  platform: 'android' | 'ios'
+): string {
+  const runtimeVersion = config[platform]?.runtimeVersion ?? config.runtimeVersion;
+  if (!runtimeVersion) {
+    throw new Error(
+      `There is neither a value or a policy set for the runtime version on "${platform}"`
+    );
+  }
+
+  if (typeof runtimeVersion === 'string') {
+    return runtimeVersion;
+  } else if (runtimeVersion['policy'] === 'nativeVersion') {
```

Comment:
any reason not to do
```suggestion
  } else if (runtimeVersion.policy === 'nativeVersion') {
```
here and below?